src/txn/txn_rollback_to_stable.c

```diff
@@ -1401,211 +1397,191 @@ __rollback_progress_msg(WT_SESSION_IMPL *session, struct timespec rollback_start
-          S2C(session)->recovery_ckpt_snap_min, S2C(session)->recovery_ckpt_snap_max,
-          S2C(session)->recovery_ckpt_snapshot_count);
-
-    while ((ret = cursor->next(cursor)) == 0) {
-        /* Log a progress message. */
-        __rollback_progress_msg(session, rollback_timer, rollback_count, &rollback_msg_count);
-        ++rollback_count;
+          "skip rollback to stable on file %s because %s", uri,
+          addr_size == 0 ? "its checkpoint address length is 0" :
+                           "it has timestamped updates and the stable timestamp is 0");
+        return (0);
+    }
 
-        WT_ERR(cursor->get_key(cursor, &uri));
-        dhandle_allocated = perform_rts = false;
+    /*
+     * The rollback operation should be performed on this file based on the following:
+     * 1. The dhandle is present in the cache and tree is modified.
+     * 2. The checkpoint durable start/stop timestamp is greater than the rollback timestamp.
+     * 3. XXX NEEDS COMMENT ON CHECK OF prepared_updates.
+     * 4. There is no durable timestamp in any checkpoint.
+     * 5. The checkpoint newest txn is greater than snapshot min txn id.
+     */
+    WT_WITH_HANDLE_LIST_READ_LOCK(session, (ret = __wt_conn_dhandle_find(session, uri, NULL)));
 
-        /* Ignore metadata and history store files. */
-        if (strcmp(uri, WT_METAFILE_URI) == 0 || strcmp(uri, WT_HS_URI) == 0)
-            continue;
+    perform_rts = ret == 0 && S2BT(session)->modified;
 
-        if (!WT_PREFIX_MATCH(uri, "file:"))
-            continue;
+    WT_ERR_NOTFOUND_OK(ret, false);
 
-        WT_ERR(cursor->get_value(cursor, &config));
+    if (perform_rts || max_durable_ts > rollback_timestamp || prepared_updates ||
+      !durable_ts_found || has_txn_updates_gt_than_ckpt_snap) {
+        WT_ERR(__wt_session_get_dhandle(session, uri, NULL, NULL, 0));
+        dhandle_allocated = true;
 
-        /* Find out the max durable timestamp of the object from checkpoint. */
-        newest_start_durable_ts = newest_stop_durable_ts = WT_TS_NONE;
-        durable_ts_found = prepared_updates = has_txn_updates_gt_than_ckpt_snap = false;
+        __wt_verbose(session, WT_VERB_RECOVERY_RTS(session),
+          "tree rolled back with durable timestamp: %s, or when tree is modified: %s or "
+          "prepared updates: %s or when durable time is not found: %s or txnid: %" PRIu64
+          " is greater than recovery checkpoint snap min: %s",
+          __wt_timestamp_to_string(max_durable_ts, ts_string[0]),
+          S2BT(session)->modified ? "true" : "false", prepared_updates ? "true" : "false",
+          !durable_ts_found ? "true" : "false", rollback_txnid,
+          has_txn_updates_gt_than_ckpt_snap ? "true" : "false");
+        WT_ERR(__rollback_to_stable_btree(session, rollback_timestamp));
+    } else
+        __wt_verbose(session, WT_VERB_RECOVERY_RTS(session),
+          "tree skipped with durable timestamp: %s and stable timestamp: %s or txnid: %" PRIu64,
```

Comment:
Thanks, I prefixed the messages with the `uri`, pushed in 99873d8b7.

Many of the verbose messages in RTS don't mention the file that's been handled -- should they?

Reply:
Yes. When RTS operating on an individual file, the dhandle is opened on that file before it gets processed. With the dhandle is open, the file name gets printed from the dhandle automatically.

Reply:
Oh, of course! Nice, I get it.

Reply:
@kommiharibabu, in 8340f752b I backed out part of my 99873d8 changes, the ones that weren't necessary because the dhandle name will be logged in the error messages.